### ISSUE TITLE
Route fails due to missing/unknown connector

### ISSUE THREAD

**Issue by gashcrumb** [syndesisio/syndesis-integration-runtime#17]
Hope this is the right repo, created a twitter search to salesforce integration, however the resulting app doesn't seem to be able to find the component for twitter:

```
org.apache.camel.RuntimeCamelException: org.apache.camel.FailedToCreateRouteException: Failed to create route flow1: Route(flow1)[[From[twitter-search--KvYqLMKvNnN50qL0k6b:parsn... because of Failed to resolve endpoint: twitter-search--KvYqLMKvNnN50qL0k6b://parsnip?delay=600000&filterOld=true due to: No component found with scheme: twitter-search--KvYqLMKvNnN50qL0k6b
```

Is the connector ID supposed to be appended like that?

**Comment by zregvart** [comment]
Yes, currently until we fix syndesisio/syndesis-rest#630 and syndesisio/syndesis-rest#660, and I think the issue here is that the connector should be named `twitter-search-connector` not `twitter-search`, we had to rename some of the connectors (we added `-connector`) to prevent name clash with connectors shipped with Camel.

Funny we have a test that checks that the `deployment.json` is in sync with connector JSON descriptors in order to prevent issues like this, but it didn't fail.

**Comment by zregvart** [comment]
Not funny at all, it the test succeeded exactly because of the name clash :)

**Comment by zregvart** [comment]
This should be fixed with syndesisio/syndesis-rest#671
